### PR TITLE
Add new hook types for Pebble Notices (JU048)

### PR DESCRIPTION
**Pull request by benhoyt**
Add new hook types for the upcoming Pebble Notices work. See ["Juju and charms" section in JU048](https://docs.google.com/document/d/16PJ85fefalQd7JbWSxkRWn0Ye-Hs8S1yE99eW7pk8fA/edit#heading=h.mpgnfy965ixr), the Pebble Notices spec. New hook names are:

* `<container>-pebble-custom-notice`
* `<container>-pebble-change-updated`

### CODE REVIEW

File: hooks/hooks.go [hunk 1]
```diff
@@ -60,7 +60,9 @@ const (
 	// kinds represent will be prefixed by the workload/container name; for example,
 	// "mycontainer-pebble-ready".
 
-	PebbleReady Kind = "pebble-ready"
+	PebbleChangeUpdate Kind = "pebble-change-update"
+	PebbleClient       Kind = "pebble-client"
```
Comment (hpidcock) [inline hooks/hooks.go]: what is mycontainer-pebble-client?

Comment (benhoyt) [reply hooks/hooks.go]: It's a Pebble notice of type "client" (confusing name, I know). Basically a custom notice sent using `pebble notify` or the API.

Comment (benhoyt) [reply hooks/hooks.go]: Renamed to `pebble-change-updated` (note the **d**) and `pebble-client-notice` for consistency and clarity.

Comment (benhoyt) [reply hooks/hooks.go]: Per spec review discussion, we've settled on `pebble-change-updated` and `pebble-custom-notice`.

**Review by wallyworld — APPROVED** [review]

**Pull request merged**